extensions/standard-processors/tests/unit/YamlConfigurationTests.cpp

```diff
@@ -1072,3 +1075,581 @@ TEST_CASE("Test serialization", "[YamlConfiguration]") {
+    - value: "#{my_value}"
+Controller Services: []
+Process Groups: []
+Input Ports: []
+Output Ports: []
+Funnels: []
+Connections: []
+Parameter Context Name: my-context
+NiFi Properties Overrides: {}
+      )";
+
+  REQUIRE_THROWS_WITH(yaml_config.getRootFromPayload(TEST_CONFIG_YAML), "Non-sensitive parameter 'my_value' cannot be referenced in a sensitive property");
+}
+
+TEST_CASE("Parameters can be used in nested process groups", "[YamlConfiguration]") {
+  ConfigurationTestController test_controller;
+  core::YamlConfiguration yaml_config(test_controller.getContext());
+
+  static const std::string TEST_CONFIG_YAML =
+      R"(
+MiNiFi Config Version: 3
+Flow Controller:
+  name: Simple TailFile
+Parameter Contexts:
+  - id: 721e10b7-8e00-3188-9a27-476cca376978
+    name: my-context
+    description: my parameter context
+    Parameters:
+    - name: lookup.frequency
+      description: ''
+      value: 12 min
+  - id: 123e10b7-8e00-3188-9a27-476cca376456
+    name: sub-context
+    description: my sub context
+    Parameters:
+    - name: batch_size
+      description: ''
+      value: 12
+Processors:
+- id: b0c04f28-0158-1000-0000-000000000000
+  name: TailFile
+  class: org.apache.nifi.processors.standard.TailFile
+  max concurrent tasks: 1
+  scheduling strategy: TIMER_DRIVEN
+  scheduling period: 1 sec
+  auto-terminated relationships list: [success]
+  Properties:
+    Batch Size: 1
+    File to Tail: ./logs/minifi-app.log
+    Initial Start Position: Beginning of File
+    tail-mode: Single file
+    Lookup frequency: "#{lookup.frequency}"
+Controller Services: []
+Input Ports: []
+Output Ports: []
+Funnels: []
+Connections: []
+Parameter Context Name: my-context
+Process Groups:
+  - id: 2a3aaf32-8574-4fa7-b720-84001f8dde43
+    name: Sub process group
+    Processors:
+    - id: 12304f28-0158-1000-0000-000000000000
+      name: SubTailFile
+      class: org.apache.nifi.processors.standard.TailFile
+      max concurrent tasks: 1
+      scheduling strategy: TIMER_DRIVEN
+      scheduling period: 1 sec
+      auto-terminated relationships list: [success]
+      Properties:
+        Batch Size: "#{batch_size}"
+        File to Tail: ./logs/minifi-app.log
+        Initial Start Position: Beginning of File
+        tail-mode: Single file
+        Lookup frequency: 1 sec
+    Parameter Context Name: sub-context
+      )";
+
+  std::unique_ptr<core::ProcessGroup> flow = yaml_config.getRootFromPayload(TEST_CONFIG_YAML);
+  REQUIRE(flow);
+  auto* proc = flow->findProcessorByName("TailFile");
+  REQUIRE(proc);
+  CHECK(proc->getProperty("Batch Size") == "1");
+  CHECK(proc->getProperty("Lookup frequency") == "12 min");
+  auto* subproc = flow->findProcessorByName("SubTailFile");
+  REQUIRE(subproc);
+  CHECK(subproc->getProperty("Batch Size") == "12");
+  CHECK(subproc->getProperty("Lookup frequency") == "1 sec");
+}
+
+TEST_CASE("Subprocessgroups cannot inherit parameters from parent processgroup", "[YamlConfiguration]") {
```

Comment:
Is this on purpose? I thought they'd be inherited.

Reply:
Yes, this is on purpose. In NiFi parameter contexts are not inherited between process groups. It has to be explicitly selected which parameter context should be used for each process group. There is a feature in NiFi which will be implemented later to use parameter context inheritance. That allows parameter contexts to be derived from other parameter contexts and using the derived contexts allows the process group to use the parameters from the base parameter context.